From: Erik Kaneda <erik.kaneda@intel.com>
Date: Fri, 14 Feb 2020 10:47:55 -0800
Subject: ACPICA: ASL-ASL+ converter: remove function parameters from
 cv_init_file_tree()
Patch-mainline: v5.7-rc1
Git-commit: 1f57e6a2c4ae3840449ea1c2708cea43aa15f3ec
References: jsc#SLE-16407

ACPICA commit 3ba6fa55a4a32d8b6fc28f9f285506ea0e359296

These parameters can be computed inside of the function from the
Table parameter.

Link: https://github.com/acpica/acpica/commit/3ba6fa55
Signed-off-by: Erik Kaneda <erik.kaneda@intel.com>
Signed-off-by: Bob Moore <robert.moore@intel.com>
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Acked-by: Lee, Chun-Yi <jlee@suse.com>
---
 drivers/acpi/acpica/acconvert.h |    4 +---
 drivers/acpi/acpica/acmacros.h  |    4 ++--
 2 files changed, 3 insertions(+), 5 deletions(-)

--- a/drivers/acpi/acpica/acconvert.h
+++ b/drivers/acpi/acpica/acconvert.h
@@ -65,9 +65,7 @@ void cg_write_aml_comment(union acpi_par
 /*
  * cvparser
  */
-void
-cv_init_file_tree(struct acpi_table_header *table,
-		  u8 *aml_start, u32 aml_length);
+void cv_init_file_tree(struct acpi_table_header *table);
 
 void cv_clear_op_comments(union acpi_parse_object *op);
 
--- a/drivers/acpi/acpica/acmacros.h
+++ b/drivers/acpi/acpica/acmacros.h
@@ -477,7 +477,7 @@
 #define ASL_CV_PRINT_ONE_COMMENT(a,b,c,d) cv_print_one_comment_type (a,b,c,d);
 #define ASL_CV_PRINT_ONE_COMMENT_LIST(a,b) cv_print_one_comment_list (a,b);
 #define ASL_CV_FILE_HAS_SWITCHED(a)       cv_file_has_switched(a)
-#define ASL_CV_INIT_FILETREE(a,b,c)      cv_init_file_tree(a,b,c);
+#define ASL_CV_INIT_FILETREE(a)      cv_init_file_tree(a);
 
 #else
 
@@ -492,7 +492,7 @@
 #define ASL_CV_PRINT_ONE_COMMENT(a,b,c,d)
 #define ASL_CV_PRINT_ONE_COMMENT_LIST(a,b)
 #define ASL_CV_FILE_HAS_SWITCHED(a)       0
-#define ASL_CV_INIT_FILETREE(a,b,c)
+#define ASL_CV_INIT_FILETREE(a)
 
 #endif
 
